File: sdk/clojure/src/hitechcloud/sdk/billing.clj
(ns hitechcloud.sdk.billing
  (:require [hitechcloud.sdk.http-client :as http]))

;; Billing & Contracts: balance, invoices, payment methods

(defn get_balance "GET /api/balance - Get account balance and credit" [client & [params]] (http-get client "/api/balance" params))

(defn list_invoices "GET /api/invoice - List all invoices" [client & [params]] (http-get client "/api/invoice" params))

(defn get_invoice "GET /api/invoice/{id} - Get invoice details" [client & [params]] (http-get client "/api/invoice/{id}" params))

(defn apply_credit "POST /api/invoice/{id}/credit - Apply account credit to invoice" [client & [params]] (http-post client "/api/invoice/{id}/credit" params))

(defn get_payment_methods "GET /api/payment - List available payment methods" [client & [params]] (http-get client "/api/payment" params))

(defn get_payment_fees "GET /api/payment/fees - List payment methods with fees" [client & [params]] (http-get client "/api/payment/fees" params))

File: sdk/fsharp/src/HiTechCloudSDK/Billing.fs
namespace HiTechCloud.SDK

/// Billing & Contracts: balance, invoices, payment methods
type BillingResource(httpClient: HttpClient) =

    /// GET /api/balance - Get account balance and credit
    member this.GetBalance() =
        async {
            return! httpClient.getAsync("/api/balance")
        }

    /// GET /api/invoice - List all invoices
    member this.ListInvoices() =
        async {
            return! httpClient.getAsync("/api/invoice")
        }

    /// GET /api/invoice/{id} - Get invoice details
    member this.GetInvoice() =
        async {
            return! httpClient.getAsync("/api/invoice/{id}")
        }

    /// POST /api/invoice/{id}/credit - Apply account credit to invoice
    member this.ApplyCredit() =
        async {
            return! httpClient.postAsync("/api/invoice/{id}/credit")
        }

    /// GET /api/payment - List available payment methods
    member this.GetPaymentMethods() =
        async {
            return! httpClient.getAsync("/api/payment")
        }

    /// GET /api/payment/fees - List payment methods with fees
    member this.GetPaymentFees() =
        async {
            return! httpClient.getAsync("/api/payment/fees")
        }

